src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -334,6 +310,162 @@ public CaseDTO getCaseByCaseReference(final long caseRef, CaseQueryRequestDTO re
+    cachedCase.setCaseType(caseType);
+    cachedCase.setEstabType(modifyRequestDTO.getEstabType().getCode());
+    cachedCase.setAddressType(caseType.name());
+    cachedCase.setAddressLine1(modifyRequestDTO.getAddressLine1());
+    cachedCase.setAddressLine2(modifyRequestDTO.getAddressLine2());
+    cachedCase.setAddressLine3(modifyRequestDTO.getAddressLine3());
+    cachedCase.setCeOrgName(modifyRequestDTO.getCeOrgName());
+    dataRepo.writeCachedCase(cachedCase);
+  }
+
+  private void sendAddressModifiedEvent(
+      UUID originalCaseId, ModifyCaseRequestDTO modifyRequestDTO, CaseContainerDTO caseDetails) {
+    CollectionCaseCompact collectionCase =
+        CollectionCaseCompact.builder()
+            .id(originalCaseId)
+            .caseType(modifyRequestDTO.getCaseType().name())
+            .ceExpectedCapacity(modifyRequestDTO.getCeUsualResidents())
+            .build();
+    AddressCompact originalAddress = caseDTOMapper.map(caseDetails, AddressCompact.class);
+    AddressCompact newAddress = caseDTOMapper.map(caseDetails, AddressCompact.class);
+
+    newAddress.setAddressLine1(modifyRequestDTO.getAddressLine1());
+    newAddress.setAddressLine2(modifyRequestDTO.getAddressLine2());
+    newAddress.setAddressLine3(modifyRequestDTO.getAddressLine3());
+    newAddress.setEstabType(modifyRequestDTO.getEstabType().getCode());
+    newAddress.setOrganisationName(modifyRequestDTO.getCeOrgName());
+
+    AddressModification payload =
+        AddressModification.builder()
+            .collectionCase(collectionCase)
+            .originalAddress(originalAddress)
+            .newAddress(newAddress)
+            .build();
+    sendEvent(EventType.ADDRESS_MODIFIED, payload, originalCaseId);
+  }
+
+  private void sendAddressTypeChangedEvent(
+      UUID updatedCaseId, UUID originalCaseId, ModifyCaseRequestDTO modifyRequestDTO) {
+    CollectionCase collectionCase = new CollectionCase();
+    collectionCase.setId(originalCaseId.toString());
+    collectionCase.setCeExpectedCapacity(modifyRequestDTO.getCeUsualResidents());
+    collectionCase.setContact(null);
+
+    Address address = new Address();
+    address.setAddressLine1(modifyRequestDTO.getAddressLine1());
+    address.setAddressLine2(modifyRequestDTO.getAddressLine2());
+    address.setAddressLine3(modifyRequestDTO.getAddressLine3());
+    address.setEstabType(modifyRequestDTO.getEstabType().getCode());
+    address.setOrganisationName(modifyRequestDTO.getCeOrgName());
+    address.setAddressType(modifyRequestDTO.getCaseType().name());
+
+    collectionCase.setAddress(address);
+
+    AddressTypeChanged payload =
+        AddressTypeChanged.builder()
+            .newCaseId(updatedCaseId)
+            .collectionCase(collectionCase)
+            .build();
+    sendEvent(EventType.ADDRESS_TYPE_CHANGED, payload, updatedCaseId);
+  }
+
+  private void updateModifiedResponse(CaseDTO response, ModifyCaseRequestDTO modifyRequestDTO) {
```

Comment:
perhaps better named? is update not same as modify? suggest prepareModificationResponse()

Reply:
done